apps/www/registry/default/ui/carousel.tsx

```diff
@@ -194,6 +193,36 @@ const CarouselItem = React.forwardRef<
 })
 CarouselItem.displayName = "CarouselItem"
 
+const CarouselDots = React.forwardRef<
+  HTMLDivElement,
+  React.HTMLAttributes<HTMLDivElement>
+>(({ className, ...props }) => {
+  const { selectedIndex, scrollTo, api } = useCarousel()
+  return (
+    <div
+      className={cn(
+        "embla__dots mt-4 flex w-[calc(100vw-16px)] items-center justify-center gap-4",
+        className
+      )}
+      {...props}
+    >
+      {api?.scrollSnapList().map((_, index) => (
+        <Button
+          key={index}
+          className={cn(
+            "embla__dot h-4 w-2 rounded-full",
+            index === selectedIndex
+              ? "embla__dot--selected"
+              : "bg-primary-foreground"
+          )}
+          onClick={() => scrollTo(index)}
```

Comment:
The expression **onClick={() => scrollTo(index)}** creates a new function object every time, which could lead to performance issues. 

How about defining it beforehand like 
**const handleScrollTo = (index) => () => scrollTo(index);** 

and then using it as **onClick={handleScrollTo(index)}?**